nixos/modules/tasks/network-interfaces-systemd.nix

```diff
@@ -92,6 +86,40 @@ in
             };
           };
         });
+        networks."99-ethernet-default-dhcp" = lib.mkIf cfg.useDHCP {
```

Comment:
This was actually placed in the wrong section: the `99-ethernet-default-dhcp.network` only gets added if `cfg.interfaces != {}` since it's in the `forEach`-expression of `cfg.interfaces`.

I fixed it in the mayflower fork in https://github.com/mayflower/nixpkgs/commit/6d68cdf6d96030e6f7e1a85c48eb57def29177fa